Twitch/Twitch.Base/Services/NewAPI/NewTwitchAPIServiceBase.cs

```diff
@@ -105,6 +107,65 @@ public async Task<IEnumerable<T>> GetPagedDataResultAsync<T>(string requestUri,
             return results;
         }
 
+        /// <summary>
+        /// Performs a GET REST request using the provided request URI for New Twitch API-wrapped data that has array of segments.
+        /// </summary>
+        /// <param name="requestUri">The request URI to use</param>
+        /// <param name="maxResults">The maximum number of results. Will be either that amount or slightly more</param>
+        /// <returns>A type-casted object of the contents of the response</returns>
+        public async Task<T> GetPagedDataSegmentsResultAsync<T>(string requestUri, int maxResults = 1)
```

Comment:
This logic for this method is very specific to handling this API (no indication that "segments" is not a unique thing to stream schedules), so I would like to avoid hard-coding that logic in to here and instead put it into the dedicated API method to handle it. I would re-work this to keep it as simple & generic as possible:

- Remove the name "Segments" from the method and change it to something like "GetPagedSingleDataResultAsync".
- Instead of it performing the logic to do the page looping directly, have it pass the for the cursor.
- 100 won't work for a maxResults here (The stream API only allows up to 25), so remove the default value of 1 to force a value to be provided. This will instead of it being the maxResults for multiple queries, this is now a single-query maxResults.
- Update the return type to return the NewTwitchAPISingleDataRestResult<T> object type, rather than the T from it
- Update the calling method to simply combine the results, determine if it reached the total maxResults, and continue querying if necessary.